lib/resourcebuilder/apps.go

```diff
@@ -74,6 +75,14 @@ func waitForDeploymentCompletion(ctx context.Context, client appsclientv1.Deploy
 		if d.Generation <= d.Status.ObservedGeneration && d.Status.UpdatedReplicas == d.Status.Replicas && d.Status.UnavailableReplicas == 0 {
 			return true, nil
 		}
+		if conditionsLength > 0 {
```

Comment:
Can we aggregate error from all `False` conditions?

Reply:
@abhinavdahiya updated per our conversation earlier today.

[DeploymentConditionType](https://github.com/kubernetes/kubernetes/blob/39fc3c49564aa346a39202c1a5b84a9c150ca43b/pkg/apis/apps/types.go#L454-L467)

Reply:
nit: Can we aggregate the status of failure to a single error Print.
It can be multi-line.